### DISCUSSION TITLE
Issue with 'finn_design' Module Not Found During Synthesis in Vivado

### DISCUSSION THREAD

**Discussion by Nishith0402** [Xilinx/finn#1145] [Q&A]
Hi,

I am currently working on a project involving a FINN IP module, which was generated through the FINN compiler and included in a new Vivado project. The project block design includes the following components: ZYNQ7 PS, AXI DMA, AXI4 STREAM DATA FIFO, AXI stream data width converter, and the FINN IP module.

I have encountered a synthesis error indicating that the **'finn_design'** module is not found. The error is reported in the **design_1_finn_design_0_0.v** file, even though the **finn_design.v** file is already uploaded into the project sources directory in Vivado(Red rectangular box) . I have included repositories of FINN IP and its sub-modules as well.

Below are the relevant screenshots for reference:

**Block Design:**

![Screenshot from 2024-07-30 18-38-22](https://github.com/user-attachments/assets/4937dbd5-7dbb-42f0-8b31-1af2f3dcaaf4)

**Synthesis Error:** 

![Screenshot from 2024-07-30 18-39-40](https://github.com/user-attachments/assets/241fc45d-12f2-4a23-ab0f-eb17c4984202)

**Project Hierarchy:**

![WhatsApp Image 2024-07-30 at 7 14 42 PM](https://github.com/user-attachments/assets/76310044-afce-47df-a0e0-7189435ba41f)

Could you please provide guidance on resolving this issue? Any help would be greatly appreciated.

Thank you!

**Comment by auphelia** [comment]
Hi @Nishith0402 ,
This tutorial might be helpful: https://github.com/Xilinx/finn/tree/dev/tutorials/fpga_flow
From your description I cannot tell, but we sometimes also see issues when moving the FINN IP to a different location/server for integration into a bigger design. Because there will be some internal paths set, e.g. to the weight files and these paths break when the design is moved.
